addons/ColorPreview/ColorPreview.gd

```diff
@@ -262,37 +266,45 @@ func on_picker_popup_close() -> void:
 	if not hovering_line:
 		return
 	var text := current_textedit.get_line(hovering_line)
-	var color_match : RegExMatch = match_color_in_string(text)
+	var color_match : RegExMatch = regex.search(text)
 	var new_color = get_line_color(current_textedit, hovering_line)
 
 	if color_from_regex_match(color_match) == new_color:
 		return # don't change if equal -> doesn't mess up constants, strings etc.
 
-	if color_match.get_string('const'): # replace the whole color string
+	if color_match.get_string("const"): # replace the whole color string
 		text = text.replace(color_match.get_string(), "Color" + str(new_color))
 
 	else: # only replace inside parenthesis to cover Color(...) and vec4(...)
 		var color_string = str(new_color).replace("(", "").replace(")", "")
-		text = text.replace(color_match.get_string("params"), color_string)
+		var params := color_match.get_string("params1")
+		if params == "":
+			color_match.get_string("params2")
```

Comment:
does this have any effect?

Reply:
The regex match results (`color_match`) contains now the all three regex results and they all have different names ("const", "params1", "params2"), so this code checks first which one, "params1" or "params2", has something. The old code could use one name ("params") for two regexes, so the old code could always get the result from "params".

Reply:
yep. but the return value of `get_string ` isn't used

Reply:
Oops, I'll fix that.